### PR TITLE
Admin api

### PR DESCRIPTION
**Pull request by KattyaCuevas**
- Refactor of application_controller and request model
- Add requests route and controller to see the list of requests
- Create a new request when the request from the user is accepted

### CODE REVIEW

File: Api/config/routes.rb [hunk 1]
```diff
@@ -1,21 +1,18 @@
 Rails.application.routes.draw do
   scope '/api' do
-  # sessions routes
+    # sessions routes
     post '/login', to: 'sessions#create'
     delete '/logout', to: 'sessions#destroy'
 
-    resources :users, only: [:show, :index] 
+    resources :users, only: %i[show index]
```
Comment (MariethPerez) [inline Api/config/routes.rb:7]: Please Could you explain me , why  you put `%i` before `[show index]`

File: Api/app/controllers/admin/requests_controller.rb [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    def index
+      render json: Request.manager
+    end
+
+    def update
+      @request.update(status: params[:status])
+      if @request.save
+        render json: @request, status: :ok
+      else
+        render json: @request.errors, status: :bad_reques
```
Comment (DiegoCuevas) [inline Api/app/controllers/admin/requests_controller.rb]: Isn't it a `:bad_request`? or is it ok like this `:bad_reques`?

**Review by MariethPerez — DISMISSED** [review]

File: Api/app/controllers/admin/requests_controller.rb (hunk 2 above)
Comment (KattyaCuevas) [reply Api/app/controllers/admin/requests_controller.rb]:
```suggestion
        render json: @request.errors, status: :bad_request
```

**Comment by sergiodxa** [comment]
This is how the flow of the status will work with the latest changes @KattyaCuevas will commit.

```
Pending
  -> Agreed
    -> Accepted
    -> Rejected
  -> Disagreed
```

The initial status of a request is `Pending`, in this step the request will be visible only for the Requester and the Requested. The Requested Front Desk agent could _agree_ or _disagree_ with the request, if the new status is `Disagreed` the flow finish there and nothing happens, but if the status is `Agree` it will be visible for the Manager.

The manager will now have the option to _accept_ or _reject_ a workshift change, if the status changes to `Accepted` it will update the workshifts of the Requester and Requested but if it changes to `Disagreed` then the flow will finish there and nothing will happen as with `Disagreed`.

This way you don't need a `rol` for the request, just use the `status` to define who sees what requests.

File: Api/config/routes.rb (hunk 1 above)
Comment (KattyaCuevas) [reply Api/config/routes.rb:7]: It's a literal object, `%i` means an array of symbols. So `[:show, :index]` is the same of `%i[show index]`

**Review by DiegoCuevas — APPROVED** [review]

File: Api/config/routes.rb (hunk 1 above)
Comment (MariethPerez) [reply Api/config/routes.rb:7]: Thanks Kattya for your explanation

**Review by MariethPerez — APPROVED** [review]

**Pull request merged**